Lib/pathlib.py

```diff
@@ -664,8 +560,9 @@ def relative_to(self, *other):
         else:
             to_abs_parts = to_parts
         n = len(to_abs_parts)
-        cf = self._flavour.casefold_parts
-        if (root or drv) if n == 0 else cf(abs_parts[:n]) != cf(to_abs_parts):
+        def nc(parts):
```

Comment:
Why the function defined here? I would either abstract it out, move it to the top with some whitespace, or better yet just expand the code so it isn't so compact as that ternary operator within an `if` guard is really hard to follow (too many `if` in a single line).